fuel-asm/src/op.rs

```diff
@@ -0,0 +1,33 @@
+//! Definitions and implementations for each unique instruction type, one for each
+//! unique `Opcode` variant.
+use super::{GMArgs, GTFArgs, Imm12, Imm18, Instruction, RegId};
+
+// Here we re-export the generated instruction types and constructors, but extend them with
+// `gm_args` and `gtf_args` short-hand constructors below to take their `GMArgs` and `GTFArgs`
+// values respectively.
+#[doc(inline)]
+pub use super::_op::*;
+
+impl GM {
+    /// Construct a `GM` instruction from its arguments.
+    pub fn from_args(ra: RegId, args: GMArgs) -> Self {
+        Self::new(ra, Imm18::new(args as _))
+    }
+}
+
+impl GTF {
+    /// Construct a `GTF` instruction from its arguments.
+    pub fn from_args(ra: RegId, rb: RegId, args: GTFArgs) -> Self {
+        Self::new(ra, rb, Imm12::new(args as _))
+    }
+}
+
+/// Construct a `GM` instruction from its arguments.
+pub fn gm_args(ra: u8, args: GMArgs) -> Instruction {
+    Instruction::GM(GM::from_args(RegId::from(ra), args))
```

Comment:
should we be validating that the RegId is within a valid range (0..64)?

Reply:
Done in c8b6ad1